vg/draw/draw_test.go

```diff
@@ -0,0 +1,58 @@
+package draw
+
+import (
+	"image/color"
+	"reflect"
+	"testing"
+
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/recorder"
+)
+
+func TestCrop(t *testing.T) {
+	ls := LineStyle{
+		Color: color.NRGBA{0, 20, 0, 123},
+		Width: 0.1 * vg.Inch,
+	}
+
+	c1 := NewCanvas(recorder.New(96), 6, 3)
+	c11 := c1.Crop(0, 0, -3, 0)
+	c12 := c1.Crop(3, 0, 0, 0)
+
+	c2 := NewCanvas(recorder.New(96), 6, 3)
+	c21 := Canvas{
+		Canvas: c2.Canvas,
+		Rectangle: Rectangle{
+			Min: Point{0, 0},
+			Max: Point{3, 3},
+		},
+	}
+	c22 := Canvas{
+		Canvas: c2.Canvas,
+		Rectangle: Rectangle{
+			Min: Point{3, 0},
+			Max: Point{6, 3},
+		},
+	}
+	if c11.Rectangle != c21.Rectangle {
+		t.Errorf("%#v != %#v", c11.Rectangle, c21.Rectangle)
+	}
+	if c12.Rectangle != c22.Rectangle {
+		t.Errorf("%#v != %#v", c11.Rectangle, c21.Rectangle)
+	}
+
+	c11.StrokeLine2(ls, c11.Min.X, c11.Min.Y,
+		c11.Min.X+3*vg.Inch, c11.Min.Y+3*vg.Inch)
+	c12.StrokeLine2(ls, c12.Min.X, c12.Min.Y,
+		c12.Min.X+3*vg.Inch, c12.Min.Y+3*vg.Inch)
+	c21.StrokeLine2(ls, c21.Min.X, c21.Min.Y,
+		c21.Min.X+3*vg.Inch, c21.Min.Y+3*vg.Inch)
+	c22.StrokeLine2(ls, c22.Min.X, c22.Min.Y,
+		c22.Min.X+3*vg.Inch, c22.Min.Y+3*vg.Inch)
+
+	c1Actions := c1.Canvas.(*recorder.Canvas).Actions
+	c2Actions := c2.Canvas.(*recorder.Canvas).Actions
+	if !reflect.DeepEqual(c1Actions, c2Actions) {
```

Comment:
After you have r1 and r2, this becomes `!reflect.DeepEqual(r1.Actions, r2.Actions)` and the 2 lines above go away.